hack/setup-mac-kind.sh

```diff
@@ -0,0 +1,283 @@
+#!/usr/bin/env bash
+
+# Copyright 2021 Chainguard, Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+#if [[ -z "${GITHUB_WORKSPACE}" ]]; then
+#  echo "This script is expected to run in the context of GitHub Actions."
+#  exit 1
+#fi
+
+# Defaults
+K8S_VERSION="v1.21.x"
+KNATIVE_VERSION="1.1.0"
+REGISTRY_NAME="registry.local"
+REGISTRY_PORT="5000"
+CLUSTER_SUFFIX="cluster.local"
+NODE_COUNT="1"
+
+while [[ $# -ne 0 ]]; do
+  parameter="$1"
+  case "${parameter}" in
+    --k8s-version)
+      shift
+      K8S_VERSION="$1"
+      ;;
+    --knative-version)
+      shift
+      KNATIVE_VERSION="$1"
+      ;;
+    --registry-url)
+      shift
+      REGISTRY_NAME="$(echo "$1" | cut -d':' -f 1)"
+      REGISTRY_PORT="$(echo "$1" | cut -d':' -f 2)"
+      ;;
+    --cluster-suffix)
+      shift
+      CLUSTER_SUFFIX="$1"
+      ;;
+    *) echo "unknown option ${parameter}"; exit 1 ;;
+  esac
+  shift
+done
+
+# The version map correlated with this version of KinD
+KIND_VERSION="v0.11.1"
+case ${K8S_VERSION} in
+  v1.19.x)
+    K8S_VERSION="1.19.11"
+    KIND_IMAGE_SHA="sha256:07db187ae84b4b7de440a73886f008cf903fcf5764ba8106a9fd5243d6f32729"
+    KIND_IMAGE="kindest/node:${K8S_VERSION}@${KIND_IMAGE_SHA}"
+    ;;
+  v1.20.x)
+    K8S_VERSION="1.20.7"
+    KIND_IMAGE_SHA="sha256:cbeaf907fc78ac97ce7b625e4bf0de16e3ea725daf6b04f930bd14c67c671ff9"
+    KIND_IMAGE="kindest/node:${K8S_VERSION}@${KIND_IMAGE_SHA}"
+    ;;
+  v1.21.x)
+    K8S_VERSION="1.21.1"
+    KIND_IMAGE_SHA="sha256:69860bda5563ac81e3c0057d654b5253219618a22ec3a346306239bba8cfa1a6"
+    KIND_IMAGE="kindest/node:${K8S_VERSION}@${KIND_IMAGE_SHA}"
+    ;;
+  *) echo "Unsupported version: ${K8S_VERSION}"; exit 1 ;;
+esac
+
+#############################################################
+#
+#    Install KinD
+#
+#############################################################
+echo '::group:: Install KinD'
+
+# Disable swap otherwise memory enforcement does not work
+# See: https://kubernetes.slack.com/archives/CEKK1KTN2/p1600009955324200
+#sudo swapoff -a
+#sudo rm -f /swapfile
+
+# Use in-memory storage to avoid etcd server timeouts.
+# https://kubernetes.slack.com/archives/CEKK1KTN2/p1615134111016300
+# https://github.com/kubernetes-sigs/kind/issues/845
+sudo mkdir -p /tmp/etcd
+#sudo mount -t tmpfs tmpfs /tmp/etcd
+
+curl -Lo ./kind "https://github.com/kubernetes-sigs/kind/releases/download/${KIND_VERSION}/kind-$(uname)-amd64"
```

Comment:
I was testing with arm64 when I was doing local testing, maybe change to arm64?

Reply:
or maybe change this to:
kind-$(uname)-$(uname -m)

Reply:
teh things i commented out related to the mapped drive to tmpfs which there is no concept of in MacOS i.e. https://superuser.com/questions/448585/is-there-an-implementation-of-tmpfs-for-mac-os-x

i removed the tmpfs mount and the mounted drives for the kind nodes

Reply:
Yeah, I had to make those changes as well :) However, is your mac intel or arm based?

Reply:
m1 (apple silcon)

Reply:
ok, well, if it worked for you as is, let's get it in, I'll retry it on my setup :)